HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,714 @@
+- **Token reject**: An additional transaction allowing any user to send a token balance back to the token treasury without paying custom fess
+
+### Receiver’s Automatic Token Association Choice: Zero, Limited, or Unlimited
+
+Every account in Hedera may have zero or more token associations. An association indicates that the account is capable of holding tokens of the associated type. For example, if an account is associated with a fungible Fruit token, then it can hold a balance of Fruit tokens. If an account is associated with a non-fungible Space Wombats token, then it can hold any number of Space Wombats. If the account does not have a Fruit association, or a Space Wombat association, then it can hold neither of those kinds of tokens.
+
+Accounts have a property, `max_auto_associations` that keeps track of the maximum number of automatic associations allowed by the account. Internally, the node software keeps track of automatic and manual associations separately, so that manually associating does not reduce the number of available auto association *slots*. Today, the user pays for each `max_auto_associations` slot, even if there is not an association. The value for `max_auto_associations` is limited to non-negative numbers. `0` means no automatic associations are allowed. If `max_auto_associations` is a positive number, then only that many automatic associations are permitted. A separate property, `used_auto_associations`, keeps track of how many automatic associations were used. If `max_auto_associations` is some number, and then later is switched to a number N, such that N is < `used_auto_associations`, then no new automatic associations are possible. Manual associations are always possible, regardless of the values of `max_auto_associations` and  `used_auto_associations`. This current logic is confusing and not fully transparent.
+
+This HIP changes this behavior in the following ways:
+
+- The value `-1` is now permitted for `max_auto_associations`, and this value means “unlimited”.
+- The account holder no longer prepays for unused slots regardless of the setting for `max_auto_associations`. Slots are only paid for when used, and are initially paid for by the sender during automatic associations. The account is responsible for covering rent for each association after the first auto-renewal period. `used_auto_associations` is incremented every time an auto-association is made.
+- The default value for automatically-created accounts will now be `-1`. This means, if an account is created automatically (by performing a token transfer to an alias that does not yet exist, see [HIP-583](https://hips.hedera.com/hip/hip-583)), then the automatically created account is configured with unlimited automatic token associations. Accounts created via the normal flow will continue to have a `used_auto_associations` default of 0.
```

Comment:
Suggestion for improvement.

The default value for `used_auto_associations` for **new** automatically-created accounts will now be `-1`. This means, if an account is created automatically (by performing a token transfer to an alias that does not yet exist, see [HIP-583](https://hips.hedera.com/hip/hip-583)), then the automatically created account is configured with unlimited automatic token associations. Accounts created by using the [AccountCreateTransaction](https://docs.hedera.com/hedera/sdks-and-apis/sdks/accounts-and-hbar/create-an-account) (HAPI: [CryptoCreate](https://github.com/hashgraph/hedera-protobufs/blob/main/services/crypto_create.proto))
will continue to have a `used_auto_associations` default of 0. Accounts created before this feature is released will remain unchanged.

Reply:
Thanks, great suggestion, will adopt.
One correction is `max_auto_associations` instead.
`used_auto_associations` will always start as 0 and is a node internal property not part of the API, only mentioned to better explain the counter tracking logic.